### DISCUSSION TITLE
Correct configuration for using the build cache instrumentation hook for file system

### DISCUSSION THREAD

**Discussion by ptpittman** [caching-tools/next-shared-cache#797] [General]
I'm running a a nextjs 14 app dir app, pre-building approximately 1000 pages at deployment time. I'm trying to set things up with the new experimental.instrumentationHook file to bring these pre-builds over into the custom redis cache. 

Without the cache handler set, at build time these are stored on disk as you'd expect in .next/cache/fetch-cache. However, when I set my custom cache handler to my ever so slightly tweaked version of the documented redis handler, regardless of whether the instrumentation hook is also set, this cache is simply _not_ created at build time. With the custom handler enabled, the only folders in .next/cache are "swc" and "webpack" after build completes.

Is there something I'm meant to do in the cache handler to ensure the fetch cache gets written to disk?

At the end of my cache-handler.js, I'm returning:

```
  return {
    handlers: [redisHandler, LRUHandler],
    useFileSystem: true,
  };
```

But that useFileSystem doesn't seem to do anything. Perhaps I have it in the wrong place? Posting this as a discussion and not a bug, as I'm sure this is probably user error....

Thanks for any help!

**Comment by better-salmon** [comment]
@ptpittman hello! What version of the lib do you use?

**Reply by ptpittman** [reply]
Hi! I was using 1.7.4 when I posted this, I can bump to 1.8.1 just now and give it another go, were there changes that might address this?

**Reply by ptpittman** [reply]
Update: same behaviour with 1.8.1 and next 14.2.15.

Also tried, just for curiosity, using ``npm run build`` rather than my usual ``pnpm build``, but same output.

Only anomaly I can really see in output of the build is one of these is thrown early on:

```
[CacheHandler] Next.js does not use the cache in development mode. Use production mode to enable caching.
```

Even though this is a build phase, and the cache handler is explicitly only called as:

```
  if (process.env.NODE_ENV == "production") {
    nextConfig.cacheHandler = require.resolve("./cache-handler.js");
  }
```

The process.env.NEXT_PHASE variable mentioned in a couple of places appears to be undefined.
